core/engine/src/builtins/temporal/mod.rs

```diff
@@ -254,7 +254,8 @@ pub(crate) fn to_relative_temporal_object(
         JsValue::Object(relative_to_obj) => Some(relative_to_obj.into()),
         _ => None,
     }
-    .map(|plane_date| Ok::<_, JsError>(to_temporal_date(&plane_date, None, context)?.inner))
+    .map(|plain_date| Ok::<_, JsError>(to_temporal_date(&plain_date, None, context)))
```

Comment:
Thought: The previous match could probably be rewritten to use an early return on `None`. It would simplify this part by a lot.